docs/docs/tutorials/web_socket_counter.md

```diff
@@ -0,0 +1,410 @@
+---
+sidebar_position: 5
+title: 🔌 WebSocket Counter
+description: Build a real time "Counter" powered by WebSockets.
+---
+
+# WebSocket Counter 🔌
+
+:::info
+**Difficulty**: 🟠 Intermediate<br/>
+**Length**: 30 minutes
+
+Before getting started, [read the prerequisites](/docs/overview#prerequisites) to make sure your development environment is ready.
+:::
+
+## Overview
+
+In this tutorial, we're going to build an app that exposes a single endpoint which handles WebSocket connections and maintains a real-time counter which can be incremented/decremented by connected clients.
+
+When we're done, we should be able to connect to the `/ws` endpoint and send/receive messages.
+
+```dart
+import 'package:web_socket_channel/web_socket_channel.dart';
+
+void main() async {
+  final channel = WebSocketChannel.connect(Uri.parse('ws://localhost:8080/ws'));
+  channel.stream.listen(print);
+
+  channel.sink.add('__increment__');
+  channel.sink.add('__decrement__');
+
+  channel.sink.close();
+}
+```
+
+And we should see the following output:
+
+```
+0 # initial
+1 # increment
+0 # decrement
+```
+
+## Creating a new app
+
+To create a new Dart Frog app, open your terminal, `cd` into the directory where you'd like to create the app, and run the following command:
+
+```bash
+dart_frog create web_socket_counter
+```
+
+You should see an output similar to:
+
+```
+✓ Creating web_socket_counter (0.1s)
+✓ Installing dependencies (1.7s)
+
+Created web_socket_counter at ./web_socket_counter.
+
+Get started by typing:
```

Comment:
is this supposed to be markdown? I think we're missing some backticks

Reply:
No it's supposed to be the output from the terminal.